app/controllers/notifications_controller.rb

```diff
@@ -18,24 +18,42 @@ def index
 
     guardian.ensure_can_see_notifications!(user)
 
+    if notification_types = params[:filter_by_types]&.split(",")&.map(&:to_sym).presence
```

Comment:
```suggestion
    if notification_types = params[:filter_by_types]&.split(",").presence
```

The extra `Array#map` is unnecessary and allocates an extra array. The `to_sym` can be called when we're looping through again below.